tests/test-cases/Group6-VIC-Machine/6-04-Create-Basic.robot

```diff
@@ -210,32 +262,48 @@ Create VCH - Existing VCH name
+    Log  ${output}
 
-    # # Create dummy VM
-    # ${rc}  ${output}=  Run And Return Rc And Output  govc vm.create -net=%{PUBLIC_NETWORK} %{VCH-NAME}
-    # Should Be Equal As Integers  ${rc}  0
+    # VCH creation should succeed on ESXi
+    Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Get Docker Params  ${output}  ${true}
+    Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Should Contain  ${output}  Installer completed successfully
+    Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Log To Console  Installer completed successfully: %{VCH-NAME}
+    Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Run Keyword And Ignore Error  Cleanup VIC Appliance On Test Server
+
+    # VCH creation should fail on VC
+    Run Keyword If  '%{HOST_TYPE}' == 'VC'  Should Contain  ${output}  The name '%{VCH-NAME}' already exists.
+
+    Cleanup Dummy VM And VCH Folder
+    Cleanup VCH Bridge Network  %{BRIDGE_NETWORK}
+
+Create VCH - Existing VM Name At Folder Location
+    Set Test Environment Variables
+    Run Keyword And Ignore Error  Cleanup Dangling VMs On Test Server
+    Run Keyword And Ignore Error  Cleanup Datastore On Test Server
 
-    # ${output}=  Run  bin/vic-machine-linux create --name=%{VCH-NAME} --target="%{TEST_USERNAME}:%{TEST_PASSWORD}@%{TEST_URL}" --thumbprint=%{TEST_THUMBPRINT} --image-store=%{TEST_DATASTORE} --bridge-network=%{BRIDGE_NETWORK} --public-network=%{PUBLIC_NETWORK} ${vicmachinetls}
-    # Log  ${output}
+    # This case cannot occur on standalone ESXi's
+    Pass Execution If  '%{HOST_TYPE}' == 'ESXi'  ESXi does not support folders, skipping test.
 
-    # # VCH creation should succeed on ESXi
-    # Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Get Docker Params  ${output}  ${true}
-    # Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Should Contain  ${output}  Installer completed successfully
-    # Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Log To Console  Installer completed successfully: %{VCH-NAME}
-    # Run Keyword If  '%{HOST_TYPE}' == 'ESXi'  Run Keyword And Ignore Error  Cleanup VIC Appliance On Test Server
+    # setup environment
+    Create Dummy VM In Place Of The VCH Folder
 
-    # # VCH creation should fail on VC
-    # Run Keyword If  '%{HOST_TYPE}' == 'VC'  Should Contain  ${output}  The name '%{VCH-NAME}' already exists.
+    ${output}=  Run  bin/vic-machine-linux create --name=%{VCH-NAME} --target="%{TEST_USERNAME}:%{TEST_PASSWORD}@%{TEST_URL}" --thumbprint=%{TEST_THUMBPRINT} --image-store=%{TEST_DATASTORE} --bridge-network=%{BRIDGE_NETWORK} --public-network=%{PUBLIC_NETWORK} ${vicmachinetls}
+    Log  ${output}
+    Should Contain  ${output}  The name '%{VCH-NAME}' already exists.
 
-    # ${rc}  ${output}=  Run And Return Rc And Output  govc vm.destroy %{VCH-NAME}
-    # Should Be Equal As Integers  ${rc}  0
-    # Run Keyword If  %{DRONE_BUILD_NUMBER} != 0  Cleanup VCH Bridge Network
+    ${rc}=  Run And Return Rc  govc vm.destroy %{VCH-NAME}
```

Comment:
I think you are leaking in the datastore still though?

Reply:
doesn't this need to use your cleanup dummy vm and vch folder keyword too?

Reply:
In this case there shouldn't be a VCH folder. since the vm exists at the exact path where we would create it. So I just make a call to destroy the vm instead. When calling the destroy on the vm it should delete all of its configuration files as well. Since no vch was created there shouldn't be anything else in the datastore.